### DISCUSSION TITLE
Override a provider family

### DISCUSSION THREAD

**Discussion by derolf** [rrousselGit/riverpod#2510] [Q&A]
I am having a family provider generated by riverpod-generator and it doesn't have an "override" function.

What's the proper way to override a family provider?

**Comment by rrousselGit** [comment]
It's coming soon, in theory at least.

**Reply by IlutschiI** [reply]
Is it coming though? 👀

**Reply by maxence-futures** [reply]
any update ?

**Comment by derolf** [comment]
Any workaround until then?

**Reply by derolf** [reply]
I have the following workaround -- basically wrapping the implementation into a delegate:

```
@riverpod
Foo family(FamilyRef ref, Bar key) => ref.watch(familyDelegateProvider)(ref, key);

@riverpod
Foo Function(FamilyRef ref, Bar key) familyDelegate(FamilyDelegateRef _) => (ref, key) {
  // actual implementation
}
```
